modules/virtual_machine/main.tf

```diff
@@ -77,12 +78,12 @@ resource "azurerm_virtual_machine" "this" {
   os_profile {
     computer_name  = var.name
     admin_username = var.username
-    admin_password = var.password
+    admin_password = local.disable_password_authentication == true ? null : var.password
     custom_data    = var.custom_data
   }
 
   os_profile_linux_config {
-    disable_password_authentication = var.password == null ? true : false
+    disable_password_authentication = local.disable_password_authentication
```

Comment:
I would add here some `lifecycle.precondition` - it's possible now to set the `password` to `null` and not specify any SSH keys. At least one should be set.

this. can probably apply to the scale set as well

Reply:
Added lifecycle to both vmss and virtual machine in which you `var.password` and `var.ssh_keys` cannot be null at the same time :

```
lifecycle {
      precondition {
      condition = var.password != null || var.ssh_keys != null
      error_message = "Either password or ssh_keys must be set in order to have access to the device"
  }
}
```